src/app/features/room/RoomInput.tsx

```diff
@@ -289,25 +288,41 @@ export const RoomInput = forwardRef<HTMLDivElement, RoomInputProps>(
 
       if (plainText === '') return;
 
-      let body = plainText;
-      let formattedBody = customHtml;
-      if (replyDraft) {
-        body = parseReplyBody(replyDraft.userId, trimReplyFromBody(replyDraft.body)) + body;
-        formattedBody =
-          parseReplyFormattedBody(
-            roomId,
-            replyDraft.userId,
-            replyDraft.eventId,
-            replyDraft.formattedBody
-              ? trimReplyFromFormattedBody(replyDraft.formattedBody)
-              : sanitizeText(replyDraft.body)
-          ) + formattedBody;
-      }
+      const body = plainText;
+      const formattedBody = customHtml;
 
       const content: IContent = {
         msgtype: msgType,
         body,
       };
+      const userIdMentions = new Set<string>();
+      if (replyDraft) {
+        userIdMentions.add(replyDraft.userId);
+      }
+      let mentionsRoom = false;
+      editor.children.forEach((node: Descendant): void => {
+        if("type" in node && node.type !== "paragraph") return;
+        const paragraph: ParagraphElement = node as ParagraphElement;
+        paragraph.children?.forEach((child: InlineElement): void => {
+          if ("type" in child && child.type === "mention") {
+            const mention: MentionElement = child as MentionElement;
+            if (mention.name === "@room" && !mention.id.startsWith("@")) {
+              mentionsRoom = true
+            } else {
+              userIdMentions.add(mention.id)
+            }
```

Comment:
please compare the id with current roomid/alias instead of checking for @room as it can be changed. Also as room mention are also `MentionElement` it is good the check `isUserId` so roomIdOrAlias doesn't end up into `userIdMentions`